### PR TITLE
refactor(api): Remove pipette pairing

### PR DESCRIPTION
**Pull request by sfoster1**
Pipette pairing was an experimental feature developed for making the
OT-2 higher throughput by simultaneously executing liquid handling
commands on both pipettes. The problem is that this requires the
pipettes to be extremely exactly physicall positioned relative to each other,
which basically isn't possible - they have to be exactly 36mm apart to
comply with the pitch of tipracks and ANSI wells, and that isn't
consistent enough. In the meantime, supporting the feature adds overhead
and complexity to every liquid handling code path.

This commit removes the (undocumented) feature from the protocol API and the hardware controller.

### CODE REVIEW

**Review by amitlissack — APPROVED** [review]
I don't see anything weird. There's a fair amount of unit/integration test around all this code.

**Review by DerekMaggio — APPROVED** [review]
Well that is way simpler.

File: api/src/opentrons/protocol_api_experimental/pipette_context.py [hunk 1]
```diff
@@ -362,8 +361,3 @@ def return_height(self) -> float:  # noqa: D102
     @property
     def well_bottom_clearance(self) -> Clearances:  # noqa: D102
         raise NotImplementedError()
-
-    def pair_with(  # noqa: D102
```
Comment (mcous) [inline api/src/opentrons/protocol_api_experimental/pipette_context.py:366]: Nice

**Review by mcous — APPROVED** [review]
10/10 would review again

**Pull request merged**